src/Transform/ONNX/ConstProp.cpp

```diff
@@ -614,6 +616,131 @@ class ConstPropSplitV11Pattern : public OpRewritePattern<ONNXSplitV11Op> {
   }
 };
 
+// https://github.com/onnx/onnx/blob/master/docs/Changelog.md#ScatterND-13
+/*
+ * output = np.copy(data)
+ * update_indices = indices.shape[:-1]
+ * for idx in np.ndindex(update_indices):
+ *     output[indices[idx]] = updates[idx]
+ */
+template <typename T>
+LogicalResult ScatterNDImpl(
+    PatternRewriter &rewriter, ONNXScatterNDOp scatterNdOp, char *raw_buffer) {
+
+  char *data_value = getArrayFromAttributeOrBuffer(
+      rewriter, scatterNdOp.data().getDefiningOp());
+  char *indices_value = getArrayFromAttributeOrBuffer(
+      rewriter, scatterNdOp.indices().getDefiningOp());
+  char *updates_value = getArrayFromAttributeOrBuffer(
+      rewriter, scatterNdOp.updates().getDefiningOp());
+
+  auto data_shape = scatterNdOp.data().getType().cast<ShapedType>().getShape();
+  auto indices_shape =
+      scatterNdOp.indices().getType().cast<ShapedType>().getShape();
+  auto updates_shape =
+      scatterNdOp.updates().getType().cast<ShapedType>().getShape();
+
+  // the output shape keep same with data, so fill with input data temporarily
+  T *output_data = reinterpret_cast<T *>(data_value);
+  int64_t *indices_data = reinterpret_cast<int64_t *>(indices_value);
+  T *updates_data = reinterpret_cast<T *>(updates_value);
+
+  int32_t n_slices = 1;
+  int32_t slice_size = 1;
+
+  int32_t outer_dims = indices_shape.size() - 1;
+  int32_t indices_nd = indices_shape[outer_dims];
+  int32_t updates_dims = updates_shape.size();
+
+  for (int32_t i = 0; i < outer_dims; i++) {
+    n_slices *= indices_shape[i];
+  }
+
+  for (int32_t i = outer_dims; i < updates_dims; ++i) {
+    slice_size *= updates_shape[i];
+  }
+
+  auto flat_size = [](ArrayRef<int64_t> &shape) -> int64_t {
```

Comment:
I don't understand why we need to declare a lambda function here. It is used once only. Could you elaborate on this? 

Just in case you would like to compute the number of elements given a shape, there is a function named `getNumOfElements`.

Reply:
I will use getNumberOfElements() instead of the lambda, thanks!